pkg/sql/schema_changer.go

```diff
@@ -187,14 +187,11 @@ func isPermanentSchemaChangeError(err error) bool {
 		return false
 	case *pgerror.Error:
 		switch err.Code {
-		case pgerror.CodeSerializationFailureError, pgerror.CodeConnectionFailureError:
+		case pgerror.CodeSerializationFailureError:
 			return false
 
 		case pgerror.CodeInternalError:
-			if err.Message == context.DeadlineExceeded.Error() {
-				return false
-			}
-
+			return false
```

Comment:
Why are you removing this? Otherwise, a single deadline exceeded in a schema change will roll back the whole thing.

Reply:
The original code was returning false (not rolling back a schema change) only for an internal error with message deadline exceeded. The new code now doesn't rollback a schema change for *any* internal error.

Reply:
Ooh, okay. That seems dangerous too. Aren't there lots of internal errors that will never resolve themselves? That's why we have rollbacks.